Cubical/Algebra/CommRing/FGIdeal.agda

```diff
@@ -12,140 +12,90 @@ open import Cubical.Foundations.Powerset
+  isLinearCombination- V = map λ α → (λ i → - α .fst i) , cong (-_) (α .snd) ∙ sym (dist- _ V)
 
   {- 0r is the trivial linear Combination -}
-  isLinearCombination0 : {n : ℕ} (l : Vec R n)
-                        → isLinearCombination l 0r
-  isLinearCombination0 l = ∣ replicate 0r , sym (dist0 _ l) ∣
+  isLinearCombination0 : {n : ℕ} (V : FinVec R n)
+                       → isLinearCombination V 0r
+  isLinearCombination0 V = ∣ _ , sym (dist0 V) ∣
 
   {- Linear combinations are stable under left multiplication -}
-  isLinearCombinationL· : {n : ℕ} (l : Vec R n)
-                        → (r x : R)
-                        → isLinearCombination l x
-                        → isLinearCombination l (r · x)
-  isLinearCombinationL· l r x =
-    elim (λ _ → isPropPropTrunc)
-         λ {(cx , px) →
-            ∣ map (r ·_) cx ,
-             (r · _                               ≡[ i ]⟨ r · px i ⟩
-              r · linearCombination cx l          ≡⟨ step1 _ cx l r  ⟩
-              linearCombination (map (r ·_) cx) l ∎) ∣}
-    where
-      step2 : (r c a t1 : R) → r · (c · a + t1) ≡ r · (c · a) + r · t1
-      step2 = solve Ring
-      step3 : (r c a t1 : R) → r · (c · a) + t1 ≡ r · c · a + t1
-      step3 = solve Ring
-      step1 : (k : ℕ) (cx l : Vec R k)
-              → (r : R)
-              → r · linearCombination cx l ≡ linearCombination (map (r ·_) cx) l
-      step1 ℕ.zero [] [] r = 0RightAnnihilates _
-      step1 (ℕ.suc k) (c ∷ cx) (a ∷ l) r =
-        r · (c · a + linearCombination cx l)               ≡⟨ step2 r c a _ ⟩
-        r · (c · a) + r · linearCombination cx l           ≡[ i ]⟨ r · (c · a) + step1 _ cx l r i ⟩
-        r · (c · a) + linearCombination (map (_·_ r) cx) l ≡⟨ step3 r c a _ ⟩
-        r · c · a + linearCombination (map (_·_ r) cx) l ∎
-
-  generatedIdeal : {n : ℕ} → Vec R n → IdealsIn Ring
-  generatedIdeal l = makeIdeal Ring
-                               (λ x → isLinearCombination l x , isPropPropTrunc)
-                               (isLinearCombination+ l)
-                               (isLinearCombination0 l)
-                               λ r → isLinearCombinationL· l r _
+  isLinearCombinationL· : {n : ℕ} (V : FinVec R n) (r : R) {x : R}
+                        → isLinearCombination V x → isLinearCombination V (r · x)
+  isLinearCombinationL· V r = map λ α → (λ i → r · α .fst i) , cong (r ·_) (α .snd)
+                                                            ∙∙ ∑Mulrdist r (λ i → α .fst i · V i)
+                                                            ∙∙ ∑Ext λ i → ·Assoc r (α .fst i) (V i)
+
+  generatedIdeal : {n : ℕ} → FinVec R n → IdealsIn Ring
+  generatedIdeal V = makeIdeal Ring
+                               (λ x → isLinearCombination V x , isPropPropTrunc)
+                               (isLinearCombination+ V)
+                               (isLinearCombination0 V)
+                               λ r → isLinearCombinationL· V r
+
+-- bad notation?
```

Comment:
What does it denote?

Reply:
It's supposed to be the ideal generated by a certain FinVec, but you have to make the Ring explicit anyways...